<?xml version="1.0"?>
<launch>
<!-- Load URDF on to ROS parameter server -->
    <param name="robot_description" textfile="$(find cse400_robot)/urdf/robot.urdf"/>
    
    <node name="serial_node" pkg="rosserial_python" type="serial_node.py" >
    <param name = "port" value = "/dev/ttyACM0"/>
    </node>
    
 <!-- Robot state publisher node -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type = "robot_state_publisher"/>
    
    
 <!-- Joint Controller Parameters -->
    <rosparam file ="$(find cse400_robot)/config/joint_control_params.yaml" command = "load" />
    
 <!-- Hardware Interface node -->
    <node name = "hw_interface" pkg = "cse400_robot" type = "hw_interface_node" output = "screen"/>
    
 <!-- Controller Spawner -->
    <node name = "controller_spawner" pkg = "controller_manager" type = "spawner" output = "screen"
        args = "/joint_state_controller/joint_state_pub
        /diff_drive_controller/mobile_base_controller" />

<!-- Run the code to convert IMU messages and LIDAR messages from teensy format to ROS format-->
    <node name= "teensy_to_ros" pkg = "cse400_robot" type = "teensy_to_ros.py"/>

    
<!-- Load Robot Localization Parameters on the parameter server-->
    <rosparam file="$(find cse400_robot)/config/localization_params.yaml"  command = "load" />
        
<!-- Robot Localization -->
    <node name="ekf_global" pkg="robot_localization" type="ekf_localization_node" />
    
<!-- Run the map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find cse400_robot)/map/map_config.yaml" />

    
    <node name="rviz" pkg="rviz" type="rviz"/>
    
</launch>
